docs/team/josiahkhoo.md

```diff
@@ -64,26 +112,340 @@ Click on the following links to view the code that I have contributed:
+
+Step 3. The parsed `Export` command is executed. The current filtered flashcard list is exported to `out.json`, located in the `/export/` directory.
+
+The following sequence diagram shows how the export operation works as a whole:
+
+![ExportSequenceDiagram](../images/ExportSequenceDiagram.png)
+
+The following activity diagram summarizes what happens when a user executes an `Export` command:
+
+![ExportActivityDiagram](../images/ExportActivityDiagram.png)
+
+#### Design consideration:
+
+##### Aspect: How to output the export file
+
+* **Alternative 1 (current choice):** Predefined directory of `/export/`
+  * Pros: Easy to implement.
+  * Cons: The user will have to navigate to his `/export/` folder to retrieve output file.
+
+* **Alternative 2:** User specifies which directory to save the export file to.
+  * Pros: More control over where the export file will end up at.
+  * Cons: Difficult to implement.
+  * Cons: Command becomes more complicated as the entire path needs to be typed out.
+
+_{more aspects and alternatives to be added}_
+
+### Import Feature
+
+#### Implementation
+
+The import mechanism is similarly facilitated by `Storage` and `QuickCache`. `Storage` is used to interact with the users local data, and a new `QuickCache` containing the data to be imported is read by `Storage` from local data.
+
+Given below is an example usage scenario and how the export mechanism behaves at each step.
+
+Step 1. The user places the file `in.json` that he wants to import in his `/import/`folder.
+
+Step 2. The user inputs the `import in.json` command. The following sequence diagram shows how the input command gets parsed:
+
+![ImportParserSequenceDiagram](../images/ImportParserSequenceDiagram.png)
+
+Step 3. The parsed `Import` command is executed. The flashcards from the file `in.json` is imported into his local `QuickCache`. If a flashcard has been imported before, it will not be imported again. The check for repitive flashcards is carried out using `Model#hasFlashcard` and `Flashcard#equals`.
+
+The following sequence diagram shows how the import operation works as a whole:
+
+![ImportSequenceDiagram](../images/ImportSequenceDiagram.png)
+
+The following activity diagram summarizes what happens when a user executes an `Import` command:
+
+![ImportActivityDiagram](../images/ImportActivityDiagram.png)
+
+#### Design consideration:
+
+##### Aspect: How to import the input file
+
+* **Alternative 1 (current choice):** Predefined directory of `/import/`
+  * Pros: Easy to implement.
+  * Cons: The user will have to navigate to his `/import/` folder to a place the input file in it.
+
+* **Alternative 2:** User specifies which directory to save the export file to.
+  * Pros: More control over where the import file can be from e.g. user's download folder.
+  * Cons: Difficult to implement.
+  * Cons: Command becomes more complicated as the entire path needs to be typed out.
+
+_{more aspects and alternatives to be added}_
```

Comment:
Empty line needed at the end!

Reply:
Thanks for pointing that out! Completely didn't see it lol